From 6ad8bddd4ce4e760936b4c203ca6df2ea7e5ac9a Mon Sep 17 00:00:00 2001
From: Minghuan Lian <Minghuan.Lian@freescale.com>
Date: Mon, 8 Jun 2015 09:45:40 +0800
Subject: [PATCH 188/451] dts/ls1043a: add MSI dts node

Signed-off-by: Minghuan Lian <Minghuan.Lian@freescale.com>
Signed-off-by: Mingkai Hu <Mingkai.Hu@freescale.com>
[Xulin: Original patch taken from
Linux-LS1043A-SDK-V0.4-SOURCE-20150826-yocto.iso]
Signed-off-by: Xulin Sun <xulin.sun@windriver.com>
---
 arch/arm64/boot/dts/freescale/fsl-ls1043a.dtsi |   39 ++++++++++++++++++++----
 1 files changed, 33 insertions(+), 6 deletions(-)

diff --git a/arch/arm64/boot/dts/freescale/fsl-ls1043a.dtsi b/arch/arm64/boot/dts/freescale/fsl-ls1043a.dtsi
index a391492..fbf7fd7 100644
--- a/arch/arm64/boot/dts/freescale/fsl-ls1043a.dtsi
+++ b/arch/arm64/boot/dts/freescale/fsl-ls1043a.dtsi
@@ -78,6 +78,33 @@
 		interrupts = <1 9 0xf08>;
 	};
 
+	msi1: msi-controller1@1571000 {
+		compatible = "fsl,1s1043a-msi";
+		reg = <0x0 0x1571000 0x0 0x4>,
+		      <0x0 0x1571004 0x0 0x4>;
+		reg-names = "msiir", "msir";
+		msi-controller;
+		interrupts = <0 116 0x4>;
+	};
+
+	msi2: msi-controller2@1572000 {
+		compatible = "fsl,1s1043a-msi";
+		reg = <0x0 0x1572000 0x0 0x4>,
+		      <0x0 0x1572004 0x0 0x4>;
+		reg-names = "msiir", "msir";
+		msi-controller;
+		interrupts = <0 126 0x4>;
+	};
+
+	msi3: msi-controller3@1573000 {
+		compatible = "fsl,1s1043a-msi";
+		reg = <0x0 0x1573000 0x0 0x4>,
+		      <0x0 0x1573004 0x0 0x4>;
+		reg-names = "msiir", "msir";
+		msi-controller;
+		interrupts = <0 160 0x4>;
+	};
+
 	tzasc: tzasc@1500000 {
 		reg = <0x0 0x1500000 0x0 0x10000>;
 		interrupts = <0 93 0x4>;
@@ -819,9 +846,8 @@
 		       0x40 0x00000000 0x0 0x00002000>; /* configuration space */
 		reg-names = "regs", "config";
 		interrupts = <0 118 0x4>, /* controller interrupt */
-			     <0 116 0x4>, /* MSI interrupt */
 			     <0 117 0x4>; /* PME interrupt */
-		interrupt-names = "intr", "msi", "pme";
+		interrupt-names = "intr", "pme";
 		num-atus = <6>;
 		#address-cells = <3>;
 		#size-cells = <2>;
@@ -830,6 +856,7 @@
 		bus-range = <0x0 0xff>;
 		ranges = <0x81000000 0x0 0x00000000 0x40 0x00010000 0x0 0x00010000   /* downstream I/O */
 			  0x82000000 0x0 0x40000000 0x40 0x40000000 0x0 0x40000000>; /* non-prefetchable memory */
+		msi-parent = <&msi1>;
 		#interrupt-cells = <1>;
 		interrupt-map-mask = <0 0 0 7>;
 		interrupt-map = <0000 0 0 1 &gic 0 110 0x4>,
@@ -844,9 +871,8 @@
 		       0x48 0x00000000 0x0 0x00002000>; /* configuration space */
 		reg-names = "regs", "config";
 		interrupts = <0 128 0x4>,
-			     <0 126 0x4>,
 			     <0 127 0x4>;
-		interrupt-names = "intr", "msi", "pme";
+		interrupt-names = "intr", "pme";
 		num-atus = <6>;
 		#address-cells = <3>;
 		#size-cells = <2>;
@@ -855,6 +881,7 @@
 		bus-range = <0x0 0xff>;
 		ranges = <0x81000000 0x0 0x00000000 0x48 0x00010000 0x0 0x00010000   /* downstream I/O */
 			  0x82000000 0x0 0x40000000 0x48 0x40000000 0x0 0x40000000>; /* non-prefetchable memory */
+		msi-parent = <&msi2>;
 		#interrupt-cells = <1>;
 		interrupt-map-mask = <0 0 0 7>;
 		interrupt-map = <0000 0 0 1 &gic 0 120  0x4>,
@@ -869,9 +896,8 @@
 		       0x50 0x00000000 0x0 0x00002000>; /* configuration space */
 		reg-names = "regs", "config";
 		interrupts = <0 162 0x4>,
-			     <0 160 0x4>,
 			     <0 161 0x4>;
-		interrupt-names = "intr", "msi", "pme";
+		interrupt-names = "intr", "pme";
 		num-atus = <6>;
 		#address-cells = <3>;
 		#size-cells = <2>;
@@ -880,6 +906,7 @@
 		bus-range = <0x0 0xff>;
 		ranges = <0x81000000 0x0 0x00000000 0x50 0x00010000 0x0 0x00010000   /* downstream I/O */
 			  0x82000000 0x0 0x40000000 0x50 0x40000000 0x0 0x40000000>; /* non-prefetchable memory */
+		msi-parent = <&msi3>;
 		#interrupt-cells = <1>;
 		interrupt-map-mask = <0 0 0 7>;
 		interrupt-map = <0000 0 0 1 &gic 0 154 0x4>,
-- 
1.7.5.4

